### PR TITLE
chore: refactor workspace members api

### LINKED ISSUE #17890: [Task - EE]: Update Service flows and API w.r.t. UserGroups
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- `getWorkspaceMembers` should also return `UserGroup` in the list
- `updatePermissionGroupForMember` should handle UserGroup
- `changeRoleAssociation` should be able to handle duplicate permissions from same workspace

### PR DESCRIPTION
**Pull request by nsarupr**
## Description

> Refactor `getWorkspaceMember` service and api call to introduce a new DTO class to be used in the EE version.

Fixes #17890 

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

> Since no functionality has changed, this doesn't require any new test case.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/WorkspaceMemberInfoDTO.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import com.appsmith.server.dtos.ce.WorkspaceMemberInfoCE_DTO;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
+
+@Setter
+@Getter
+public class WorkspaceMemberInfoDTO extends WorkspaceMemberInfoCE_DTO {
+    String userGroupId;
```
Comment (trishaanand) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/WorkspaceMemberInfoDTO.java:11]: Please remove these from CE repository. This file should not contain any code which is not useful in CE but is required in EE.

**Review by trishaanand — COMMENTED** [review]
LGTM. Minor comment added.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/WorkspaceMemberInfoDTO.java (hunk 1 above)
Comment (nsarupr) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/WorkspaceMemberInfoDTO.java]: Updated the Pojo

**Review by trishaanand — APPROVED** [review]

**Pull request merged**